src/backend/distributed/shared_library_init.c

```diff
@@ -457,8 +484,13 @@ _PG_init(void)
 	set_rel_pathlist_hook = multi_relation_restriction_hook;
 	get_relation_info_hook = multi_get_relation_info_hook;
 	set_join_pathlist_hook = multi_join_restriction_hook;
+#if PG_VERSION_NUM >= PG_VERSION_18
+	ExecutorStart_hook = citus_executor_start_adapter;
+	ExecutorRun_hook = citus_executor_run_adapter;
+#else
 	ExecutorStart_hook = CitusExecutorStart;
 	ExecutorRun_hook = CitusExecutorRun;
+#endif
```

Comment:
since you are using adapters, you can pass the `#if PG_VERSION_NUM >= PG_VERSION_18` check inside the adapter.

Reply:
updated